dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/ThreadPoolController.java

```diff
@@ -1285,9 +1293,13 @@ synchronized String evaluateInterval() {
                 return "monitoring paused";
             }
 
```

Comment:
Should the above code be updated?
```
    if (poolSize < coreThreads) {
            return "poolSize < coreThreads";
            return "poolSize < coreThreads";
        }
    }
```
This ties into the reset though as well and your logic about things not adjusting between intervals.  It is a question that may have the answer of no.  If there is a short lived hang that doesn't require us to grow to the new hang pool size because it resolves with less than that and we never grow to that limit.

Reply:
That check is intended as a start-of-pool-life sanity check. The pool behavior is to start a new thread for every task until it gets to coreThreads size, and until it gets there we don't need to make any adjustments. I think it is correct to leave this one at coreThreads, rather than currentMinimumThreads. If we raised currentMinimumThreads in a hang-resolution action and didn't get there yet, we will figure that out later when resolveHang calls checkTargetPoolSize, and take action if the situation persists.